app/src/main/java/com/github/sdpteam15/polyevents/view/activity/TimeTableActivity.kt

```diff
@@ -0,0 +1,515 @@
+            }
+            startActivity(intent)
+        }
+
+        //Clone the current constraints of the constraintLayout
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+
+        //Compute the margin top to have the view start at the right position to indicate start time
+        val marginTop = ((start.minute / 60.0) * hourSizeDp).toInt()
+
+        //Constraints for the time
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.LEFT,
+            idLine,
+            ConstraintSet.LEFT,
+            0
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.RIGHT,
+            idLine,
+            ConstraintSet.RIGHT,
+            0
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.TOP,
+            idLine,
+            ConstraintSet.BOTTOM,
+            marginTop.dpToPixelsInt(this)
+        )
+        constraintSet.applyTo(constraintLayout)
+    }
+
+    /**
+     * Adds the line that indicates the current time
+     */
+    private fun addNowBar() {
+        val now = LocalDateTime.now()
+        val nowH = now.hour
+        val nowM = now.minute
+        val idLine = hourToLine[nowH] ?: return
+
+        val currentId = nextId++
+        val constraintLayout = findViewById<ConstraintLayout>(R.id.id_timetable_constraintlayout)
+
+        //Create now line
+        val line = View(this)
+        line.backgroundTintList = resources.getColorStateList(R.color.red, null)
+        line.id = currentId
+
+        //Create new parameters for the line (the line)
+        val params = ViewGroup.LayoutParams(0, nowLineHeightDP.dpToPixelsInt(this))
+        line.layoutParams = params
+        line.setBackgroundColor(Color.BLUE)
+
+        constraintLayout.addView(line)
+        nowBar = line
+
+
+        val constraintSet = ConstraintSet()
+        constraintSet.clone(constraintLayout)
+        val marginTop = ((nowM / 60.0) * hourSizeDp).toInt()
+        //Constraints for the line
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.TOP,
+            idLine,
+            ConstraintSet.BOTTOM,
+            marginTop.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.LEFT,
+            idLine,
+            ConstraintSet.LEFT,
+            linepaddingLeftDP.dpToPixelsInt(this)
+        )
+        constraintSet.connect(
+            currentId,
+            ConstraintSet.RIGHT,
+            constraintLayout.id,
+            ConstraintSet.RIGHT,
+            0
+        )
+
+        //Apply all the constraints
+        constraintSet.applyTo(constraintLayout)
+
+
+        //TODO : Maybe call the runnable itself instead of recreating each time
+        val handler = Handler()
+        val task = Runnable { refreshNowBar() }
+        val period = (60 - now.second.toLong()) * 1000
```

Comment:
It is recreated every time we go through the function, so the first time it will be 1 sec and then it will take about 60sec so that the minute goes to the next in real time